bin/BuildPackages.sh

```diff
@@ -60,36 +30,52 @@ then
         "Type: cd ../pkg; ../bin/BuildPackages.sh"
 fi
 
-# CURDIR
 CURDIR="$(pwd)"
+GAPDIR="$(cd .. && pwd)"
+COLORS=yes
+PACKAGES=
+
+# If output does not go into a terminal (but rather into a log file),
+# turn of colors.
+[[ -t 1 ]] || COLORS=no
+
+while [[ "$#" -ge 1 ]]; do
+  option="$1" ; shift
+  case "$option" in
+    --with-gaproot)   GAPDIR="$1"; shift ;;
+    --with-gaproot=*) GAPDIR=${option#--with-gaproot=}; shift ;;
+    --no-color)       COLORS=no ;;
+    -*)               echo "ERROR: unsupported argument $option" ; exit 1;;
+    *)                PACKAGES="$PACKAGES $option" ;;
```

Comment:
The pkg directory (for other reasons) may contain other directories with spaces, even though they should not. 

Anyway, could you at least make a comment here that package directories are not supposed to have spaces or other special characters in their names?